crates/lib3h/src/engine/real_engine.rs

```diff
@@ -101,48 +99,88 @@ impl<'engine> RealEngine<'engine> {
         dht_factory: DhtFactory,
     ) -> Lib3hResult<Self> {
         // Create TransportMemory as the network transport
-        let network_transport = TransportWrapper::new(TransportMemory::new());
-        // Bind & create DhtConfig
-        let binding = network_transport
-            .as_mut()
-            .bind(&config.bind_url)
-            .expect("TransportMemory.bind() failed. bind-url might not be unique?");
-        let dht_config =
-            DhtConfig::with_real_engine_config(format!("{}_tId", name).as_str(), &binding, &config);
+        let mut memory_transport = GhostTransportMemory::new();
+        let mut memory_network_endpoint = Detach::new(
+            memory_transport
+                .take_parent_endpoint()
+                .expect("exists")
+                .as_context_endpoint_builder()
+                .request_id_prefix("tmem_to_child_")
+                .build::<GatewayUserData, Lib3hTrace>(),
+        );
+
+        // Bind & create this_net_peer
+        let mut gateway_ud = GatewayUserData::new();
```

Comment:
I don't think this is an invalid pattern, but it does add some readability cognitive overhead when compared to just being able to assume the "user_data" is equivalent to a mutable reference to the owning instance.

It's also a little complicated to pass in a mutable self reference while you're still in the constructor : ), but I think, for other reasons, that we're going to want to come up with an initialization pattern in our protocols whereby our constructors do no work, and we have "Initialize" RequestToActor and RequestToActorResponse types that we wait for before doing further processing.

(this can be a follow-on task)

Reply:
I will work on this whole initialization thing inside ghost_engine.  I think we can have a single constructor that takes the multiplexer, and then a bootstrap functions.

Reply:
I've rebased ghost-engine off of this branch and can make progress there...